### PR TITLE
#2672, Adding admin action logger for Elasticsuite

### PR DESCRIPTION
**Pull request by vahonc**

### CODE REVIEW

File: src/module-elasticsuite-catalog-optimizer/etc/logging.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<logging xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Logging:etc/logging.xsd">
+    <groups>
+        <group name="elasticsuite_optimizer">
+            <label translate="true">Manage Elasticsuite Search Optimizers</label>
+            <expected_models>
+                <expected_model class="Smile\ElasticsuiteCatalogOptimizer\Model\Optimizer" />
+            </expected_models>
+            <events>
+                <event controller_action="adminhtml_optimizer_save" action_alias="save" />
```
Comment (PierreGauthier) [inline src/module-elasticsuite-catalog-optimizer/etc/logging.xml:16]: Hello,
I think the prefix of the action should be the route id, not `adminhtml`
So `smile_elasticsuite_catalog_optimizer` i think
But I did not test this yet

Comment (PierreGauthier) [reply src/module-elasticsuite-catalog-optimizer/etc/logging.xml:16]: Like in the module @Magento_Invitation@ for example

Comment (rbayet) [reply src/module-elasticsuite-catalog-optimizer/etc/logging.xml:16]: Can confirmed and just tested on a EE 2.4.3-p1.
The controller action syntax is "frontName_controller_action", the frontName being defined in the module's etc/adminhtml/routes.xml

For instance for Thesaurus, the frontName is "smile_elasticsuite_thesaurus" and not the native "adminhtml" :
```
    <router id="admin">
        <route id="smile_elasticsuite_thesaurus" frontName="smile_elasticsuite_thesaurus">
            <module name="Smile_ElasticsuiteThesaurus" before="Magento_Backend" />
        </route>
    </router>
```

File: src/module-elasticsuite-catalog-optimizer/etc/logging.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<logging xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Logging:etc/logging.xsd">
+    <groups>
+        <group name="elasticsuite_optimizer">
+            <label translate="true">Manage Elasticsuite Search Optimizers</label>
```
Comment (rbayet) [inline src/module-elasticsuite-catalog-optimizer/etc/logging.xml:11]: Please add i18n files for en_US and fr_FR.
Translation for fr_FR is "Manage Elasticsuite Search Optimizers","Gérer les optimiseurs Elasticsuite"

File: src/module-elasticsuite-thesaurus/etc/logging.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<logging xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Logging:etc/logging.xsd">
+    <groups>
+        <group name="smile_elasticsuite_thesaurus">
+            <label translate="true">Manage Elasticsuite Thesaurus</label>
```
Comment (rbayet) [inline src/module-elasticsuite-thesaurus/etc/logging.xml:12]: Please add i18n files for en_US and fr_FR.
Translation for fr_FR is "Manage Elasticsuite Thesaurus","Gérer les Thesaurus Elasticsuite"

**Review by rbayet — CHANGES_REQUESTED** [review]

**Review by rbayet — APPROVED** [review]

File: src/module-elasticsuite-catalog-optimizer/etc/logging.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
```
Comment (rbayet) [inline src/module-elasticsuite-catalog-optimizer/etc/logging.xml:6]: Please use our own file header comment.

File: src/module-elasticsuite-thesaurus/etc/logging.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
```
Comment (rbayet) [inline src/module-elasticsuite-thesaurus/etc/logging.xml:7]: Please use our own file header comment.

File: src/module-elasticsuite-catalog-optimizer/etc/logging.xml (hunk 4 above)
Comment (rbayet) [reply src/module-elasticsuite-catalog-optimizer/etc/logging.xml:6]: And add the missing `<?xml version="1.0"?>` on the first line.

**Review by rbayet — CHANGES_REQUESTED** [review]

File: src/module-elasticsuite-catalog-optimizer/etc/logging.xml (hunk 4 above)
Comment (vahonc) [reply src/module-elasticsuite-catalog-optimizer/etc/logging.xml]: Done.

File: src/module-elasticsuite-thesaurus/etc/logging.xml (hunk 5 above)
Comment (vahonc) [reply src/module-elasticsuite-thesaurus/etc/logging.xml]: Done.

**Review by rbayet — APPROVED** [review]

**Pull request merged**